gcloud-java-dns/src/main/java/com/google/gcloud/dns/BatchResultImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.google.gcloud.dns;
+
+import com.google.gcloud.dns.batch.BatchResult;
+
+/**
+ * This class holds a single result of a batch call to the Cloud DNS.
+ */
+public class BatchResultImpl<T> implements BatchResult<T, DnsException> {
+
+  private T result;
+  private boolean submitted = false;
+  private DnsException error;
+
+  BatchResultImpl() {
+  }
+
+  @Override
+  public boolean submitted() {
+    return submitted;
+  }
+
+  @Override
+  public T get() throws DnsException {
+    if(!submitted()) {
+      throw new IllegalStateException("Batch has not been submitted yet");
+    }
+    if(error != null) {
+      throw error;
+    }
+    return result;
+  }
+
+  void result(T result) {
```

Comment:
those will need to be protected, so can be used by subclasses (see my comment above) in the other service packages.